splice_si_api/src/main/java/com/splicemachine/si/impl/store/IgnoreTxnSupplierImpl.java

```diff
@@ -50,12 +56,14 @@ public boolean shouldIgnore(Long txnId) throws IOException{
         boolean ignore = false;
 
         init();
-        
-        if (!cache.isEmpty()) {
-            for (Pair<Long, Long> range : cache) {
-                if (txnId > range.getFirst() && txnId < range.getSecond()) {
-                    ignore = true;
-                    break;
+
+        synchronized (this) {
```

Comment:
Can we get rid of this synchronized block? I think it should be possible to make `cache` volatile and make `refresh` atomically swap it.

Reply:
Changed in commit https://github.com/splicemachine/spliceengine/pull/5103/commits/19138b4bb9ba883e90b87b67324df4a83a7fd959